packages/create-block/lib/scaffold.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+const { dirname, join } = require( 'path' );
+const makeDir = require( 'make-dir' );
+const { readFile, writeFile } = require( 'fs' ).promises;
+const { render } = require( 'mustache' );
+
+/**
+ * Internal dependencies
+ */
+const initWPScripts = require( './init-wp-scripts' );
+const { code, clear, info, success } = require( './log' );
+const { hasWPScriptsEnabled, getOutputFiles } = require( './templates' );
+
+module.exports = async function( templateName, {
+	namespace,
+	slug,
+	title,
+	description,
+	dashicon,
+	category,
+	author,
+	license,
+	version,
+} ) {
+	clear();
+	info( `Creating a new WordPress block in "${ slug }" folder.` );
+
+	const outputFiles = getOutputFiles( templateName );
+	const view = {
+		namespace,
+		slug,
+		machineName: `${ namespace }_${ slug }`.replace( /\-/g, '_' ),
+		title,
+		description,
+		dashicon,
+		category,
+		version,
+		author,
+		license,
+		textdomain: namespace,
+	};
+
+	await Promise.all(
+		Object.keys( outputFiles ).map( async ( fileName ) => {
+			const template = await readFile(
+				join( __dirname, `templates/${ outputFiles[ fileName ] }.mustache` ),
+				'utf8'
+			);
+			const filePath = `${ slug }/${ fileName.replace( /\$slug/g, slug ) }`;
```

Comment:
Are we using this `$slug` replace convention in any of the current template offerings? Or is it for future enhancement / custom templates?

**Edit:** Okay, I think I see how this works now, where the files are named based on the `outputFiles` keys, not based on how they exist in the repository. I might wonder if we really need this, vs. just reading in all of the files in the template directory, then stripping `.mustache` suffix in whatever it outputs as.

Reply:
There is this established pattern in WordPress plugins where the name of the entry PHP file is the same as the plugin name, e.g.:
`gutenberg/gutenberg.php`

I will add a comment to make it clear.

Reply:
The main thing I was wondering is if, rather than:

```
create-block/lib/templates/esnext/style-css.mustache
```

We'd just have it as:

```
create-block/lib/templates/esnext/style.css.mustache
```

That way, we don't need this mapping at all, we just remove the `.mustache` suffix when writing the file.

I see what you mean about the plugin name, so I guess we might need the substitution in _some_ cases.

To that, I'd wonder:

- Could we just embed the `$slug` convention in the file name as well?
- Or: Can we limit the mapping to just those override cases?
- Or: Is there an option for a statically-named PHP file here, like `index.php` ? (i.e. ignore the convention)

Reply:
The first attempt in:
https://github.com/WordPress/gutenberg/pull/19867/commits/660e3f5ad6003e30d495bcc689ae5b1a75e28982

Implements:
> Could we just embed the $slug convention in the file name as well?

I will double-check whether we can go with `index.php`.